test/components/structures/TimelinePanel-test.tsx

```diff
@@ -98,68 +101,195 @@ const setupTestData = (): [MatrixClient, Room, MatrixEvent[]] => {
+                        if (setting === "sendReadReceipt") return false;
+
+                        return undefined;
+                    });
+                });
+
+                afterEach(() => {
+                    mocked(SettingsStore.getValue).mockReset();
+                });
+
+                it("should send a fully read marker and a private receipt", async () => {
+                    await renderTimelinePanel();
+                    timelineSet.addLiveEvent(ev1, {});
+                    await flushPromises();
+
+                    // @ts-ignore
+                    await timelinePanel.sendReadReceipts();
+
+                    // Expect the private reception to be sent directly
+                    expect(client.sendReadReceipt).toHaveBeenCalledWith(ev1, ReceiptType.ReadPrivate);
+                    // Expect the fully_read marker not to be send yet
+                    expect(client.sendReadReceipt).not.toHaveBeenCalledWith(ev1, ReceiptType.FullyRead, true);
+
+                    client.sendReadReceipt.mockClear();
+
+                    // @ts-ignore simulate user activity
+                    await timelinePanel.updateReadMarker();
+
+                    // It should not send the receipt again.
+                    expect(client.sendReadReceipt).not.toHaveBeenCalledWith(ev1, ReceiptType.ReadPrivate);
+                    // Expect the fully_read marker to be sent after user activity.
+                    expect(client.sendReadReceipt).toHaveBeenCalledWith(ev1, ReceiptType.FullyRead, true);
+                });
+            });
+        });
+
+        describe("and there is a thread timeline", () => {
+            const threadEv1 = new MatrixEvent({
+                event_id: "thread_ev1",
+                sender: "@u2:m.org",
+                origin_server_ts: 222,
+                type: EventType.RoomMessage,
+                content: {
+                    ...createMessageEventContent("hello 2"),
+                    "m.relates_to": {
+                        event_id: ev1.getId(),
+                        rel_type: RelationType.Thread,
+                    },
+                },
+            });
+
+            beforeEach(() => {
+                client.supportsThreads.mockReturnValue(true);
+                setUpTimelineSet(ev1);
+            });
+
+            it("should send receipts but no fully_read when reading the thread timeline", async () => {
+                await renderTimelinePanel();
+                timelineSet.addLiveEvent(threadEv1, {});
+                await flushPromises();
+
+                // @ts-ignore
+                await timelinePanel.sendReadReceipts();
+
+                // fully_read is not supported for threads per spec
+                expect(client.sendReadReceipt).not.toHaveBeenCalledWith(threadEv1, ReceiptType.FullyRead, true);
+                expect(client.sendReadReceipt).toHaveBeenCalledWith(threadEv1, ReceiptType.Read);
+            });
```

Comment:
Add a test that covers the case when the event we scrolled to is _before_ the last sent read receipt.

Reply:
See https://github.com/matrix-org/matrix-react-sdk/pull/10864/commits/2be2635e65842c5b357d0e3deb70337948a7a9c1